### PR TITLE
Add a Chevron component

### PR DESCRIPTION
**Pull request by azangru**
## Type
- New feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1161

## Description
- Added a single React component to cover all cases of our chevron usage
- Added a ShowHide component, which consists of a label associated with a chevron (used in EntityViewer; will be used more commonly)
- Added Storybook stories for Chevron and ShowHide 

## Deployment URL
http://chevron.review.ensembl.org

## Views affected
- Entity viewer
- Species page (the ExpandableSection component)
- Chevron that opens/closes the sidebar
- Help app menu
- The Accordion component

### CODE REVIEW

File: src/ensembl/src/shared/components/chevron/Chevron.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const chevronClasses = classNames(
+    styles.chevron,
+    { [styles[`chevron_${props.direction}`]]: isNonDefaultDirection },
+    { [styles.chevron_animated]: props.animateDirectionChange },
+    props.classNames?.svg
+  );
+
+  const Wrapper = props.onClick
+    ? 'button'
+    : props.classNames?.wrapper
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/chevron/Chevron.tsx:47]: how does it manage when there is both a `wrapper` class and a click handler passed as props?

File: src/ensembl/src/shared/components/chevron/Chevron.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import { ReactComponent as ChevronDown } from 'static/img/shared/chevron-down.svg';
+
+import styles from './Chevron.scss';
+
+export type Direction = 'up' | 'down' | 'left' | 'right';
+
+export type Props = {
+  direction: Direction;
+  animateDirectionChange: boolean;
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/chevron/Chevron.tsx:28]: we could just name is `animate`?

File: src/ensembl/src/shared/components/expandable-section/ExpandableSection.tsx [hunk 3]
```diff
@@ -70,7 +66,11 @@ const ExpandableSection = (props: ExpandableSectionProps) => {
     <div className={wrapperClassNames}>
       {props.expandedContent && (
         <div className={styles.toggle} onClick={toggleExpanded}>
-          <Chevron className={chevronClasses} />
+          <Chevron
+            direction={isExpanded ? 'up' : 'down'}
+            animateDirectionChange={true}
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/expandable-section/ExpandableSection.tsx:71]: as said earlier `animate={true}` sounds better. The only animation we do here is a direction change.

File: src/ensembl/src/shared/components/accordion/css/Accordion.scss [hunk 4]
```diff
@@ -10,6 +10,9 @@
 }
 
 .accordionButtonDefault {
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/accordion/css/Accordion.scss:12]: I think there is a css issue here with the new Chevron button added to Accordion
Screenshot 2021-07-01 at 11 54 37

File: src/ensembl/src/shared/components/accordion/css/Accordion.scss [hunk 5]
```diff
@@ -10,6 +10,9 @@
...
+  display: grid;
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/accordion/css/Accordion.scss:13]: I think there is a css issue here with the new Chevron button added to Accordion
Screenshot 2021-07-01 at 11 54 37

File: src/ensembl/src/shared/components/show-hide/ShowHide.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
+@import 'src/styles/common';
+
+.label {
+  color: $blue;
+  cursor: pointer;
+}
+
+.chevron {
+  margin-left: 10px;
+  height: 6px;
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/show-hide/ShowHide.scss:10]: Did we decide to reduce the size of the chevron here? 
Screenshot 2021-07-01 at 12 00 03

File: src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+  const handleClick = () => {
+    const newDirection = direction === 'up' ? 'down' : 'up';
+    setDirection(newDirection);
+  };
+
+  return (
+    <div className={styles.storyWrapper}>
+      <div className={styles.showRoom}>
+        <span className={styles.label}>{'Click the chevron →'}</span>
```
Comment (imransl) [inline src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx:125]: Why do we need a JSX expression for 'Click the chevron →' ?

File: src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+import Chevron, {
+  Direction as ChevronDirection
+} from 'src/shared/components/chevron/Chevron';
+import RadioGroup from 'src/shared/components/radio-group/RadioGroup';
+
+import styles from './Chevron.stories.scss';
+
+export const ChevronStory = () => {
+  const [direction, setDirection] = useState('down');
```
Comment (imransl) [inline src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx:27]: How about making `direction` to be of type `ChevronDirection`?

**Review by jyothishnt — COMMENTED** [review]
I have finished reviewing.

File: src/ensembl/src/shared/components/chevron/Chevron.tsx (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/shared/components/chevron/Chevron.tsx:47]: > how does it manage when there is both a wrapper class and a click handler passed as props

There are tests for this case in Chevron.test.tsx, which should answer your question 😄

File: src/ensembl/src/shared/components/chevron/Chevron.tsx (hunk 2 above)
Comment (azangru) [reply src/ensembl/src/shared/components/chevron/Chevron.tsx]: 👍 

✅

File: src/ensembl/src/shared/components/expandable-section/ExpandableSection.tsx (hunk 3 above)
Comment (azangru) [reply src/ensembl/src/shared/components/expandable-section/ExpandableSection.tsx]: ✅

File: src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx (hunk 7 above)
Comment (azangru) [reply src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx]: ✅

File: src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx (hunk 8 above)
Comment (azangru) [reply src/ensembl/stories/shared-components/chevron/Chevron.stories.tsx]: You mean `useState<ChevronDirection>`? Done. Still need to narrow down the type in the onChange function for the RadioGroup though:

image

File: src/ensembl/src/shared/components/accordion/css/Accordion.scss (hunk 5 above)
Comment (azangru) [reply src/ensembl/src/shared/components/accordion/css/Accordion.scss:13]: Updated to hide the chevron in disabled accordion elements.

File: src/ensembl/src/shared/components/show-hide/ShowHide.scss (hunk 6 above)
Comment (azangru) [reply src/ensembl/src/shared/components/show-hide/ShowHide.scss:10]: Not sure I understand...

**Review by jyothishnt — APPROVED** [review]

**Pull request merged**